common/utils/src/main/java/org/apache/spark/SparkThrowable.java

```diff
@@ -35,19 +35,29 @@
  */
 @Evolving
 public interface SparkThrowable {
-  // Succinct, human-readable, unique, and consistent representation of the error category
-  // If null, error class is not set
-  String getErrorClass();
+  /**
+   * Succinct, human-readable, unique, and consistent representation of the error condition.
+   * If null, error condition is not set.
+   */
+  String getCondition();
+
+  /**
+   * Succinct, human-readable, unique, and consistent representation of the error category.
+   * If null, error class is not set.
+   * @deprecated Use {@link #getCondition()} instead.
+   */
+  @Deprecated
+  default String getErrorClass() { return getCondition(); }
```

Comment:
I'm curious if we need to add a rule in `scalastyle-config.xml` that prohibits `getErrorClass` calls? like:

https://github.com/apache/spark/blob/44ec70f5103fc5674497373ac5c23e8145ae5660/scalastyle-config.xml#L145-L151

Reply:
I think the warnings about deprecated method is enough for now.